BinarySearchTree/utils.go

```diff
@@ -0,0 +1,9 @@
+package BinarySearchTree
+
+func keyhash(s string) int {
```

Comment:
Ideally the name should be `KeyHash` as it ensure you can access it outside of the package as well